pkg/console/subresource/route/route.go

```diff
@@ -88,6 +101,41 @@ func Validate(route *routev1.Route) (*routev1.Route, bool) {
 	return route, changed
 }
 
+func IsCustomRouteSet(operatorConfig *operatorv1.Console) bool {
+	if len(operatorConfig.Spec.Route.Hostname) != 0 {
+		return true
+	}
+	return false
+}
+
+// Check if reference for secret holding custom serving-cert is set
+func IsCustomRouteSecretSet(operatorConfig *operatorv1.Console) bool {
+	if len(operatorConfig.Spec.Route.Secret.Name) != 0 {
+		return true
+	}
+	return false
+}
+
+func ValidateCustomRoute(operatorConfig *operatorv1.Console, client routeclient.RoutesGetter) error {
+	defaultRoute, err := client.Routes(api.OpenShiftConsoleNamespace).Get(api.OpenShiftConsoleRouteName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if is := isClusterDomain(operatorConfig.Spec.Route.Hostname, defaultRoute.Spec.Host); !is {
```

Comment:
The question here is, do we trust the `default route`, or should we pull in the `Ingress Config`?

Reply:
So I think we can trust the default route, since we will be maintaining it anyways. Also when creating we are using the `default` as the router name. Otherwise we are raising a sync error.
Should be ok then ?